### PR TITLE
feat: Add size and scroll guidelines to popover documentation 

### PR DESCRIPTION
**Pull request by tanvibhakta**
## Description

Fixes https://github.com/appsmithorg/appsmith/issues/26729

Depends on # (pr)

## Type of change

- This change is a documentation update

## How Has This Been Tested?

- Manual on storybook 
- Manual on main repo
- Jest
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: packages/design-system/src/Popover/Popover.stories.tsx [hunk 1]
```diff
@@ -64,6 +64,7 @@ const PopoverContentTemplate: ComponentStory<typeof PopoverHeader> = (args) => {
   );
 };
 
+//  TODO: The popover guidelines say that popovers should have at least one focusable element in it, but this story doesn't because PopoverHeader expects children to be a string. Change this.
```
Comment (albinAppsmith) [inline packages/design-system/src/Popover/Popover.stories.tsx:67]: When are we planning for this? Can you create a task for this.

Comment (tanvibhakta) [reply packages/design-system/src/Popover/Popover.stories.tsx:67]: We haven't planned for this yet. Sure, I'll create a task.

Comment (tanvibhakta) [reply packages/design-system/src/Popover/Popover.stories.tsx:67]: We haven't planned for this yet. Sure, I'll create a task. https://github.com/appsmithorg/appsmith/issues/26844

Comment (tanvibhakta) [reply packages/design-system/src/Popover/Popover.stories.tsx:67]: https://github.com/appsmithorg/appsmith/issues/26844

**Review by albinAppsmith — APPROVED** [review]

**Pull request merged**